base/src/main/java/uk/ac/ebi/atlas/experimentimport/ExperimentDAO.java

```diff
@@ -104,59 +95,36 @@ private void addExperimentSpeciesRows(ExperimentDTO experimentDTO) {
     public void deleteExperiment(String experimentAccession) {
 
         int deletedRecordsCount = jdbcTemplate.update(DELETE_EXPERIMENT, experimentAccession);
-
-        if (deletedRecordsCount != 1) {
-            throw new IllegalArgumentException("Experiment not found for accession " + experimentAccession);
-        }
-    }
-
-    public ExperimentDTO findPublicExperiment(String experimentAccession) {
-        return findExperiment(experimentAccession, false);
+        checkState(deletedRecordsCount == 1 , "Experiment not found for accession " + experimentAccession);
     }
 
     public ExperimentDTO findExperiment(String experimentAccession, String accessKey) {
-
-        List<ExperimentDTO> experimentDTOs = jdbcTemplate.query(SELECT_EXPERIMENT_BY_ACCESSION_AND_ACCESS_KEY, new ExperimentDTOResultSetExtractor(), experimentAccession, accessKey);
-
-        return getSingleExperiment(experimentDTOs, experimentAccession);
+        return getSingleExperiment(
+                jdbcTemplate.query(SELECT_EXPERIMENT_BY_ACCESSION_AND_ACCESS_KEY, new ExperimentDTOResultSetExtractor(), experimentAccession, accessKey),
+                experimentAccession
+        );
 
 
     }
 
-    //TODO: replace this with a one parameter method findExperimentIncludingPrivate
-    public ExperimentDTO findExperiment(String experimentAccession, boolean includePrivates) {
-
-        String findExperimentQuery = includePrivates ? SELECT_EXPERIMENT_BY_ACCESSION : SELECT_PUBLIC_EXPERIMENT_BY_ACCESSION;
-        List<ExperimentDTO> experimentDTOs = jdbcTemplate.query(findExperimentQuery, new ExperimentDTOResultSetExtractor(), experimentAccession);
-
-        return getSingleExperiment(experimentDTOs, experimentAccession);
-
+    public ExperimentDTO getExperimentAsAdmin(String experimentAccession) {
+        return getSingleExperiment(
+                jdbcTemplate.query(SELECT_EXPERIMENT_AS_ADMIN_GIVEN_ONLY_ACCESSION, new ExperimentDTOResultSetExtractor(), experimentAccession),
+                experimentAccession
+        );
     }
 
-    public void updateExperiment(String experimentAccession, boolean isPrivate) {
-
-        int recordsCount = jdbcTemplate.update(UPDATE_EXPERIMENT, toString(isPrivate), experimentAccession);
+    public void setExperimentPrivacyStatus(String experimentAccession, boolean isPrivate) {
 
-        if (recordsCount == 0) {
-            throw new IllegalArgumentException("Experiment not found for accession " + experimentAccession);
-        }
+        int recordsCount = jdbcTemplate.update(UPDATE_EXPERIMENT, isPrivateAsString(isPrivate), experimentAccession);
+        checkState(recordsCount == 1 , "Experiment not found for accession " + experimentAccession );
 
     }
 
-    private String toString(boolean isPrivate) {
+    private String isPrivateAsString(boolean isPrivate) {
```

Comment:
I still wonder who thought that having the filed `private` be a `varchar(1)` instead of a `boolean` was a good idea. 😞